ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/block/BlockImportPerformance.java

```diff
@@ -25,54 +25,84 @@
+  public static String PRESTATE_RETRIEVED_EVENT_LABEL = "Pre-state retrieved";
+  public static String BLOCK_PROCESSED_EVENT_LABEL = "Block processed";
+  public static String TRANSACTION_PREPARED_EVENT_LABEL = "Transaction prepared";
+  public static String TRANSACTION_COMMITTED_EVENT_LABEL = "Transaction committed";
+  public static String IMPORT_COMPLETED_EVENT_LABEL = "Import complete";
+
+  public static String TOTAL_PROCESSING_TIME_LABEL = "Total Processing time";
+
   private final TimeProvider timeProvider;
+  private final BlockImportMetrics blockImportMetrics;
 
   private final List<Pair<String, UInt64>> events = new ArrayList<>();
   private UInt64 timeWarningLimitTimeStamp;
   private UInt64 timeAtSlotStartTimeStamp;
 
-  public BlockImportPerformance(final TimeProvider timeProvider) {
+  public BlockImportPerformance(
+      final TimeProvider timeProvider, final BlockImportMetrics blockImportMetrics) {
     this.timeProvider = timeProvider;
+    this.blockImportMetrics = blockImportMetrics;
   }
 
   public void arrival(final RecentChainData recentChainData, final UInt64 slot) {
     timeAtSlotStartTimeStamp = secondsToMillis(recentChainData.computeTimeAtSlot(slot));
     timeWarningLimitTimeStamp =
         timeAtSlotStartTimeStamp.plus(
             secondsToMillis(recentChainData.getSpec().getSecondsPerSlot(slot)).dividedBy(3));
-    addEvent("Received");
+    addEvent(RECEIVED_EVENT_LABEL);
   }
 
   public void preStateRetrieved() {
-    addEvent("Pre-state retrieved");
+    addEvent(PRESTATE_RETRIEVED_EVENT_LABEL);
   }
 
   public void postStateCreated() {
-    addEvent("Block processed");
+    addEvent(BLOCK_PROCESSED_EVENT_LABEL);
   }
 
   public void transactionReady() {
-    addEvent("Transaction prepared");
+    addEvent(TRANSACTION_PREPARED_EVENT_LABEL);
   }
 
   public void transactionCommitted() {
-    addEvent("Transaction committed");
+    addEvent(TRANSACTION_COMMITTED_EVENT_LABEL);
   }
 
   public void processingComplete(final EventLogger eventLogger, final SignedBeaconBlock block) {
-    final UInt64 importCompletedTimestamp = addEvent("Import complete");
-
-    if (importCompletedTimestamp.isGreaterThan(timeWarningLimitTimeStamp)) {
-      UInt64 previousEventTimestamp = timeAtSlotStartTimeStamp;
-      final List<String> eventTimings = new ArrayList<>();
-      for (Pair<String, UInt64> event : events) {
-        // minusMinZero because sometimes time does actually go backwards so be safe.
-        final UInt64 stepDuration = event.right().minusMinZero(previousEventTimestamp);
-        eventTimings.add(
-            event.left() + (eventTimings.isEmpty() ? " " : " +") + stepDuration + "ms");
-        previousEventTimestamp = event.right();
+    final UInt64 importCompletedTimestamp = addEvent(IMPORT_COMPLETED_EVENT_LABEL);
+
+    final List<String> lateBlockEventTimings =
+        importCompletedTimestamp.isGreaterThan(timeWarningLimitTimeStamp)
+            ? new ArrayList<>()
+            : null;
```

Comment:
I'm not really a fan of setting a `List` to `null`.  I'd probably always set this to `new ArrayList<>()` and have a separate boolean for whether we'll need to log a warning about the late block.